### PR TITLE
Simplify sphereTriangleIntersect

### PR DESCRIPTION
**Pull request by florent-lamiraux**
and add some initializations.

### CODE REVIEW

File: include/hpp/fcl/collision_data.h [hunk 1]
```diff
@@ -344,12 +344,12 @@ struct DistanceResult
...
-  DistanceResult(FCL_REAL min_distance_ = std::numeric_limits<FCL_REAL>::max()) : min_distance(min_distance_), 
-                                                                                  o1(NULL),
-                                                                                  o2(NULL),
-                                                                                  b1(NONE),
-                                                                                  b2(NONE)
+  DistanceResult(FCL_REAL min_distance_ =
+                 std::numeric_limits<FCL_REAL>::max()):
+  min_distance(min_distance_), o1(NULL), o2(NULL), b1(NONE), b2(NONE)
   {
+    Vec3f nan; nan << sqrt (-1), sqrt (-1), sqrt (-1);
```
Comment (jmirabel) [inline include/hpp/fcl/collision_data.h]: Can you use `Vec3f nan (Vec3f::Constant(std::numeric_limits<FCL_REAL>::quiet_NaN()))` ?

File: src/narrowphase/details.h [hunk 2]
```diff
@@ -349,89 +349,52 @@ namespace fcl {
       const Vec3f& center = tf1.getTranslation();
       const FCL_REAL& radius = s.radius;
       assert (radius >= 0);
-      FCL_REAL eps (std::numeric_limits<FCL_REAL>::epsilon());
-      FCL_REAL radius_with_threshold = radius + eps;
       Vec3f p1_to_center = center - P1;
       FCL_REAL distance_from_plane = p1_to_center.dot(normal);
+      Vec3f closest_point; closest_point << sqrt (-1), sqrt (-1), sqrt (-1);
```
Comment (jmirabel) [inline src/narrowphase/details.h]: Same here.

File: src/narrowphase/details.h [hunk 3]
```diff
@@ -349,89 +349,52 @@ namespace fcl {
...
-      if(has_contact)
-        {
-          FCL_REAL distance_sqr = center_to_contact.squaredNorm();
+      if (projectInTriangle (P1, P2, P3, normal, center)) {
+        closest_point = center - normal * distance_from_plane;
+        min_distance_sqr = distance_from_plane;
+      } else {
+        // Compute distance to each each and take minimal distance
+        Vec3f nearest_on_edge;
+        min_distance_sqr = segmentSqrDistance(P1, P2, center, closest_point);
```
Comment (jmirabel) [inline src/narrowphase/details.h]: Cannot `projectInTriangle` know to which edge you are the closest ?

File: src/narrowphase/details.h [hunk 4]
```diff
@@ -2213,14 +2176,15 @@ namespace fcl {
     {
```
Comment (jmirabel) [inline src/narrowphase/details.h]: I am surprised detecting collision between a plane and a capsule takes so many operations...

File: src/narrowphase/details.h (hunk 2 above)
Comment (florent-lamiraux) [reply src/narrowphase/details.h]: Yes. I will do it in a forthcoming commit.

File: src/narrowphase/details.h (hunk 4 above)
Comment (florent-lamiraux) [reply src/narrowphase/details.h]: It might be possible to make shorter, but the current implementation seems reasonable to me.

**Comment by jmirabel** [comment]
Looks good to me. You can merge if you are done with testing.

**Pull request merged**